web/quiz_views.py

```
@@ -362,85 +680,137 @@ def _process_quiz_taking(request, quiz):
+                # Loop through each question in prepared_questions
+                for question in prepared_questions:
+                    # Extract the question id
+                    question_id = str(question.get("id"))
+
+                    # Retrieve corresponding AI data
+                    ai_question_data = ai_correction_results.get("correction", {}).get(question_id, "")
+
+                    question_type = AI_data.get(question_id, "").get("question_type", "")
+                    answers_question = answers[question_id]
+
+                    if not question_type == "multiple" and not question_type == "true_false":
```

Ruff (0.8.2)

Use `question_type != "multiple"` instead of `not question_type == "multiple"`

Replace with `!=` operator

(SIM201)